### DISCUSSION TITLE
Annotation, Marginalia, Commentary

### DISCUSSION THREAD

**Discussion by micahvandegrift** [knowledgefutures/pubpub#1496] [Support]
PubPublicans (too soon?),

We're knees deep in [Futurama:Autogeddon](https://autogeddon.pubpub.org/) and our project partner (English prof.) has a really intriguing challenge. 

The original "book" was published as an iBook 🤯 but it did have the "glossary" feature, so words could be linked and a callout open upon click, giving the reading some extra context and info. Our task with Futurama is to replicate as much of the ibook as possible but in PubPub, and we're stumped on the best way to re-do those glossary callouts (see attachment). 

<img width="828" alt="Screen Shot 2021-06-29 at 4 18 26 PM" src="https://user-images.githubusercontent.com/7580842/123861892-af072400-d8f5-11eb-8d72-be604b65ff28.png">

Challenges why we don't think they fit using the Footnote or Comment feature:
- they are long
- they are media heavy
- they contain 'textual' information that should be directly attached to a word or phrase
- the comment feature connects to a user name

I can see you all rushing to reply "that's what the comments are meant to do!" but we'd like to NOT have the comment be from a person, but rather a way to categorize the marginalia as either "Key Word" or "Annotation." We checked out Frankenbook obvs and really like the tags in the comments, but could not see how they are supposed to link or index the comments by theme, unless I wasn't clicking in the right place. 

Our hacky idea is to create two new users named 1) Key Word, and @) Annotation, then log in as them, do what needs to be done, and then the text, context, and paratext will be clear as mud. But there must be a more elegant way?

**Comment by micahvandegrift** [comment]
[This is the example ](http://middleshore.electric.press/exhibits/introduction/introduction.html)they brought to us as an ideal, using [tufte-css](https://github.com/edwardtufte/tufte-css).

**Reply by memoryrushonline** [reply]
This feels less like a tooling problem and more like a question about how readers orient themselves while reading. What you’re describing sits somewhere between annotation, glossary, and paratext, and none of those map perfectly onto conventional comment systems.

One way to think about the original iBook glossary feature is that it created context without interruption. The reader didn’t leave the page, didn’t enter a social layer, and didn’t have to interpret authorship. The callout behaved more like an extension of the text than a response to it.

From a reading and learning perspective, that distinction matters. Comments tend to imply dialogue and identity, while glossary-style marginalia functions more like a temporary scaffold — something that supports understanding in the moment and then recedes.

Your hesitation about impersonating users makes sense for that reason. Once marginalia becomes tied to a “speaker,” the reader shifts modes from reading to social interpretation. That can subtly change how the information is processed.

It may help to frame the problem not as “where do these notes live,” but as “how visible should authorship and categorization be during reading.” In many scholarly contexts, marginal information works best when it feels infrastructural rather than conversational — closer to typography or layout than to commentary.

I don’t have an elegant technical solution, but conceptually I’d be cautious about anything that foregrounds identity where the original design intent was cognitive support. Preserving that quiet, contextual role may matter more than reproducing the exact mechanics of the iBook feature.

**Comment by gabestein** [comment]
So cool! Thanks for stretching PubPub to (and beyond) its limit!

I'm not sure I have a better way to do it off the top of my head, but let's see if @coahearn has thoughts.

Only thought I have is that comment labels can be filtered in the comments settings at the bottom of any pub, so it is possible to give the user that experience of looking at something via a particular viewpoint: 

<img width="778" alt="Screen Shot 2021-07-06 at 11 25 16" src="https://user-images.githubusercontent.com/639110/124642264-e5402880-de4c-11eb-8f94-d6f2dd4d8f8d.png">

Once expanded, the comments do display the labels, so users will know what they're looking at.

<img width="1043" alt="Screen Shot 2021-07-06 at 11 27 52" src="https://user-images.githubusercontent.com/639110/124642552-351eef80-de4d-11eb-8454-136d83543231.png">

So, although not perfectly satisfying, it might be slightly less awkward to use one account (Autogedden Marginalia? idk) and label the comments as such. And then at some point we could build the option for them to auto-expand as you scroll, much like the tufte example (we've been thinking about how to make better/more intuitive use of the right column for a long while now, as you know).

But I'm not sure if that's any better than your idea. Wonder if Catherine has any thoughts, though!

**Reply by micahvandegrift** [reply]
Thanks @gabestein and if its helpful design/UX feedback we had 4 people all looking at Frankenbook to try to sort the comments and didnt think to use the filter icon to filter by category. Obviously, a "Filter" button would filter things, but for some reason none of us thought to push it and we were all seriously looking for how to.

**Reply by gabestein** [reply]
That is helpful, thank you! It's pretty buried.

**Comment by coahearn** [comment]
Hmm interesting. I really like this example—and others around annotated works—because they push us to consider notes on a text (or any kind of textual apparatus) as part of the text itself and not an addition to it. Ideally, it would be cool to support a kind of production-level comment/annotation that is part of and released with the draft. 

I'm interested, though, in learning more about what exactly makes the example you provided ideal, and I'm curious about why there's so much resistance to having the glosses tied to a name. The way I always thought about it when reading a print book is that anything between the covers is tied to the author/editor on that cover. And I think having notes from an author throughout can be more inviting to readers (my hunch, too, is that it would prompt them to comment/annotate more). Annotations on PubPub, as you know, can handle long, media-heavy content tied to a specific work or phrase. So if you were to create accounts for, say, "Key Word," I'd use the annotation option over the footnote option. You could then include a url at the end of every glass that send the reader to the full "Glossary" page (or wherever). I've seen some journals use a generic "editor" account for things like this, too, which is basically what Gabe suggested. But ultimately, I'd recommend using the author's account, using our labels to identify between "Key Words" and "Annotation," and linking to a separate Glossary page. 

As a general textual design point, I think it's always best when you give the reader as much control over what they can see and when (or, said differently, how much help and interjection they want while reading). You see this play out in all kinds of different ways in print books (mostly badly, which is why multi-page footnotes are so notorious!), but the design options expand tremendously with digital books. (If you can't tell, I think annotated books are an amazing use case for PubPub.) Designs like the one you shared don't give the reader much of a say beside the hypothesis toggle, though I think we could eventually build out a way to turn the marginalia on or off if we did move to make more of the right column. One thing the original Frankebook design did well is that it made it possible for readers to say "ok, I'm interested in notes on Mary Shelley and notes on her motivations, so only show me those." And that's what they'd see while reading! There's probably more we could do to enhance our labels and filtering options toward this end, too!

**Comment by micahvandegrift** [comment]
Appreciate the responses @gabestein @coahearn! The Authors and our team will give it some mulling, and reply here with our forward motion. Onward to the PubPub-lishy future! 🥳

**Reply by maxlecalve** [reply]
Dear team at pubpub, I'm dealing with a similar issue of a project which would include marginalia. 
https://speculativerealitieslab.pubpub.org/season-1-a-day-with-prof-vajkoczy
I'm publishing several series of graphic ethnographic notes to accompany an exhibition on neurosurgical practice, and my co-curator, a neurosurgeon herself, is commenting on the texts. We really would like this to become a common object, with her marginalia (and we hope, more from the other members of the team). It looks like for now her comments are disappearing or are not appearing when we press publish, and they also do not appear when the pubs are downloaded. Any chance that this could be changed? Also, an option to keep the comments open or auto-open as we scroll down (as suggested above in this topic) would be great (on computer screens, not mobile...) Thanks for the amazing work, you are totally enabling this project.

**Reply by gabestein** [reply]
Hi Max — the default mode is that comments on draft do not get transferred to releases, because draft is often used for editorial comments. However, we have an internal tool we've built that allows transferring draft comments to releases, which we're hoping to make available to all at some point. We'd be happy to use it to transfer over comments for you. Would you like to transfer all the comments?

PDFs are a different story — there's a technical complication, which is we'd need to regenerate the PDF every time someone adds a comment. There's also a bit of a design problem, because flowing theoretically limitless annotations in a page-based format is quite difficult to do. And then there's a social problem, which is that anyone who signs up can add comments to a public release, so we'd need to build some kind of comment pdf curation system, and that just starts to feel a little bit outside the scope of a mostly web-native platform. I can see, however, maybe including any member comments that were there at the time of release by default. Would that be interesting to you? If so, I can certainly discuss it with the team to see if it would be possible.

**Reply by maxlecalve** [reply]
Hi Gabriel, 
thanks a lot, I think it would be very useful to get this internal tool for the next releases. But for now, we did manually reenter all the comments we had so far, so please do not transfer the older ones. 
We understand regarding the PDF. Including the member comments at the time of release would be great, but then we would have to rerelease each post (after the "internal tool" is working of course). That would be nice too. 
What would be truly fantastic would be if the comments of my co-curator on this, Anna Roethe, could always be opened when the pub is viewed on a computer screen so that the reader understands that this is part of the publication. 
Cheers and thanks a lot!

**Reply by maxlecalve** [reply]
Hi again, I'm still interested in that tool you have to make comments available in releases, any development on the marginalia front? Thanks!
